pdata/ptrace/ptraceotlp/response.go

```diff
@@ -8,18 +8,23 @@ import (
 
 	jsoniter "github.com/json-iterator/go"
 
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpcollectortrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1"
 	"go.opentelemetry.io/collector/pdata/internal/json"
 )
 
 // ExportResponse represents the response for gRPC/HTTP client/server.
 type ExportResponse struct {
-	orig *otlpcollectortrace.ExportTraceServiceResponse
+	orig  *otlpcollectortrace.ExportTraceServiceResponse
+	state internal.State
```

Comment:
If this is a pointer you can make changes to reflect to copied instances.

Reply:
Yeah, that's what I thought as well. It'll introduce one allocation but I think it's fine

Reply:
Migrated all the states to a pointer 👍